internal/sveltos/profile.go

```diff
@@ -50,96 +51,149 @@ type HelmChartOpts struct {
+				PlainHTTP:             hc.PlainHTTP,
+				InsecureSkipTLSVerify: hc.InsecureSkipTLSVerify,
+			},
 		}
 
-		cp.Spec = sveltosv1beta1.Spec{
-			ClusterSelector: libsveltosv1beta1.Selector{
-				LabelSelector: metav1.LabelSelector{
-					MatchLabels: matchLabels,
-				},
-			},
-			Tier:               tier,
-			ContinueOnConflict: !opts.StopOnConflict,
+		if hc.PlainHTTP {
+			// InsecureSkipTLSVerify is redundant in this case.
+			helmChart.RegistryCredentialsConfig.InsecureSkipTLSVerify = false
 		}
 
-		for _, hc := range opts.HelmChartOpts {
-			helmChart := sveltosv1beta1.HelmChart{
-				RepositoryURL:    hc.RepositoryURL,
-				RepositoryName:   hc.RepositoryName,
-				ChartName:        hc.ChartName,
-				ChartVersion:     hc.ChartVersion,
-				ReleaseName:      hc.ReleaseName,
-				ReleaseNamespace: hc.ReleaseNamespace,
-				HelmChartAction:  sveltosv1beta1.HelmChartActionInstall,
-				RegistryCredentialsConfig: &sveltosv1beta1.RegistryCredentialsConfig{
-					PlainHTTP:             hc.PlainHTTP,
-					InsecureSkipTLSVerify: hc.InsecureSkipTLSVerify,
-				},
+		if hc.Values != nil {
+			b, err := hc.Values.MarshalJSON()
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal values to JSON for service %s: %w", hc.RepositoryName, err)
 			}
 
-			if hc.PlainHTTP {
-				// InsecureSkipTLSVerify is redundant in this case.
-				helmChart.RegistryCredentialsConfig.InsecureSkipTLSVerify = false
+			b, err = yaml.JSONToYAML(b)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert values from JSON to YAML for service %s: %w", hc.RepositoryName, err)
 			}
 
-			if hc.Values != nil {
-				b, err := hc.Values.MarshalJSON()
-				if err != nil {
-					return fmt.Errorf("failed to marshal values to JSON for service (%s) in ManagedCluster: %w", hc.RepositoryName, err)
-				}
+			helmChart.Values = string(b)
+		}
 
-				b, err = yaml.JSONToYAML(b)
-				if err != nil {
-					return fmt.Errorf("failed to convert values from JSON to YAML for service (%s) in ManagedCluster: %w", hc.RepositoryName, err)
-				}
+		spec.HelmCharts = append(spec.HelmCharts, helmChart)
+	}
 
-				helmChart.Values = string(b)
-			}
+	return spec, nil
+}
 
-			cp.Spec.HelmCharts = append(cp.Spec.HelmCharts, helmChart)
-		}
-		return nil
-	})
-	if err != nil {
-		return nil, err
+func objectMeta(owner *metav1.OwnerReference) metav1.ObjectMeta {
+	obj := metav1.ObjectMeta{
+		Labels: map[string]string{
+			hmc.HMCManagedLabelKey: hmc.HMCManagedLabelValue,
+		},
 	}
 
-	if operation == controllerutil.OperationResultCreated || operation == controllerutil.OperationResultUpdated {
-		l.Info(fmt.Sprintf("Successfully %s Profile (%s/%s)", string(operation), cp.Namespace, cp.Name))
+	if owner != nil {
+		obj.OwnerReferences = []metav1.OwnerReference{*owner}
 	}
 
-	return cp, nil
+	return obj
 }
 
+// DeleteProfile deletes a Sveltos Profile object.
 func DeleteProfile(ctx context.Context, cl client.Client, namespace string, name string) error {
```

Comment:
i know that it's not directly related to this PR but here goes the same question about the need for explicit deletion of a `Profile` object, having owner references on it, does the object require it?

Reply:
Same reason as mentioned in the comment for `DeleteClusterProfile`.